### PR TITLE
2021 javascript japanese translation

### PR DESCRIPTION
**Pull request by ksakae1216**
Please review for me.
Part of #586

### CODE REVIEW

File: src/content/ja/2021/javascript.md [hunk 1]
```diff
@@ -0,0 +1,842 @@
...
+    caption="`async`と`defer`を使用しているページの割合。",
+    sheets_gid="2038121216",
+    sql_file="breakdown_of_pages_using_async_defer.sql"
+  ) }}</figcaption>
+</figure>
+
+昨年の結果では、同じスクリプトに `async` と `defer` の両方の属性を使用しているウェブサイトがあるというアンチパターンが観察されました。これは、ブラウザがサポートしている場合は `async` にフォールバックし、IE8とIE9のブラウザでは `defer` を使用するというものです。しかし、ほとんどのサイトでは、サポートされているすべてのブラウザで`async`が優先されるため、現在ではこの方法は必要ありません。このパターンでは、ページが読み込まれるまで待つのではなく、HTMLの解析が中断されます。この使用頻度は非常に高く、モバイルページの [11.4%](../2020/javascript#how-do-we-load-our-javascript) には、`async` と `defer` 属性を持つスクリプトが少なくとも1つ使用されていました。The [root causes](https://twitter.com/rick_viscomi/status/1331735748060524551?s=20) were found and an action item was also taken down to [remove such usage going forward](https://twitter.com/Kraft/status/1336772912414601224?s=20).
+
+[根本的な原因](https://twitter.com/rick_viscomi/status/1331735748060524551?s=20)を見つけ、[今後そのような使い方をしないようにする](https://twitter.com/Kraft/status/1336772912414601224?s=20)というアクションアイテムも用意されています。
+{{ figure_markup(
```
Comment (tunetheweb) [inline src/content/ja/2021/javascript.md:207]:
```suggestion
[根本的な原因](https://twitter.com/rick_viscomi/status/1331735748060524551?s=20)を見つけ、[今後そのような使い方をしないようにする](https://twitter.com/Kraft/status/1336772912414601224?s=20)というアクションアイテムも用意されています。

{{ figure_markup(
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Looks good @ksakae1216 !

Just one small missed translation

File: src/content/ja/2021/javascript.md [hunk 2]
```diff
@@ -0,0 +1,841 @@
...
+非同期型のAJAXリクエストの代わりに、同期型のリクエストがあります。リクエストをコールバックに渡すのではなく、リクエストが完了するまでメインスレッドをブロックします。
+
+しかし、パフォーマンスやユーザー体験が、低下する可能性があるため、[このような行為は推奨されません](https://developer.mozilla.org/ja/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests#synchronous_request)。また、多くのブラウザでは、このような使用方法についてすでに警告を発しています。今でも同期型のAJAXリクエストを使用しているページがどれだけあるのか、興味をそそられますね。
+
+{{ figure_markup(
+  image="usage-sync-async.png",
+  caption="モバイルページでの同期および非同期AJAXリクエストの使用について",
+  description="モバイルページにおける同期および非同期のAJAXリクエストの使用状況を示す棒グラフ。モバイルページにおける同期型リクエストの割合は2.5％、非同期型リクエストの割合は77.6％となっている。",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=189627938&format=interactive",
+  sheets_gid="183546956",
```
Comment (tunetheweb) [inline src/content/ja/2021/javascript.md:715]:
```suggestion
  sheets_gid="1113974683",
```

File: src/content/ja/2021/javascript.md [hunk 3]
```diff
@@ -0,0 +1,842 @@
...
+    </tbody>
+  </table>
+  <figcaption>{{ figure_link(
+    caption="`async`と`defer`を使用しているページの割合。",
+    sheets_gid="2038121216",
+    sql_file="breakdown_of_pages_using_async_defer.sql"
+  ) }}</figcaption>
+</figure>
+
+昨年の結果では、同じスクリプトに `async` と `defer` の両方の属性を使用しているウェブサイトがあるというアンチパターンが観察されました。これは、ブラウザがサポートしている場合は `async` にフォールバックし、IE8とIE9のブラウザでは `defer` を使用するというものです。しかし、ほとんどのサイトでは、サポートされているすべてのブラウザで`async`が優先されるため、現在ではこの方法は必要ありません。このパターンでは、ページが読み込まれるまで待つのではなく、HTMLの解析が中断されます。この使用頻度は非常に高く、モバイルページの [11.4%](../2020/javascript#how-do-we-load-our-javascript) には、`async` と `defer` 属性を持つスクリプトが少なくとも1つ使用されていました。The [root causes](https://twitter.com/rick_viscomi/status/1331735748060524551?s=20) were found and an action item was also taken down to [remove such usage going forward](https://twitter.com/Kraft/status/1336772912414601224?s=20).
```
[the remark this answers is not in the document]
Comment (ksakae1216) [reply src/content/ja/2021/javascript.md]: Thanks, I missed.

File: src/content/ja/2021/javascript.md [hunk 4]
```diff
@@ -0,0 +1,841 @@
...
+
+モバイルページの中央値は4つの非同期リクエストを行っています。ロングテールを見ると、デスクトップページの非同期リクエストの最大数は623で、最大のモバイルページの867の非同期リクエストに抜かれています！
+
+非同期型のAJAXリクエストの代わりに、同期型のリクエストがあります。リクエストをコールバックに渡すのではなく、リクエストが完了するまでメインスレッドをブロックします。
+
+しかし、パフォーマンスやユーザー体験が、低下する可能性があるため、[このような行為は推奨されません](https://developer.mozilla.org/ja/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests#synchronous_request)。また、多くのブラウザでは、このような使用方法についてすでに警告を発しています。今でも同期型のAJAXリクエストを使用しているページがどれだけあるのか、興味をそそられますね。
+
+{{ figure_markup(
+  image="usage-sync-async.png",
+  caption="モバイルページでの同期および非同期AJAXリクエストの使用について",
```
[the remark this answers is not in the document]
Comment (ksakae1216) [reply src/content/ja/2021/javascript.md:711]: Yes, I corrected.
And line 714 corrected.

**Comment by ksakae1216** [comment]
I don't push, Because Token expired.
And I have problem. Please wait.

**Comment by tunetheweb** [comment]
OK. Feel free to enter the edits as comments as if doing a review and can then accept them in GitHub. If that's easier.

**Review by tunetheweb — APPROVED** [review]
LGTM! Thanks again for your continual work on this. Hope to get this out in next day or two once the final 2021 English Chapter is merged.

**Pull request merged**